pkg/executor/join/semi_join_probe.go

```diff
@@ -170,7 +167,10 @@ func (s *semiJoinProbe) probeForLeftSideBuildHasOtherCondition(joinedChk *chunk.
 
 		for index, result := range s.selected {
 			if result {
-				meta.setUsedFlag(*(*unsafe.Pointer)(unsafe.Pointer(&s.rowIndexInfos[index].buildRowStart)))
+				candidateRow := unsafe.Pointer(&s.rowIndexInfos[index].buildRowStart)
+				if !meta.isCurrentRowUsedWithAtomic(candidateRow) {
```

Comment:
since `matchMultiBuildRows` already check this, I think we can just set the flag.

Reply:
> since `matchMultiBuildRows` already check this, I think we can just set the flag.

done